sdk/identity/azure-identity/test/ut/azure_cli_credential_test.cpp

```diff
@@ -344,6 +344,50 @@ TEST(AzureCliCredential, UnsafeChars)
   }
 }
 
+TEST(AzureCliCredential, UnsafeCharsScopeAndTenantId)
+{
+  std::string const ValidScopeButNotTenantId = "string:withcolon";
+
+  {
+    AzureCliCredentialOptions options;
+    options.TenantId = "01234567-89AB-CDEF-0123-456789ABCDEF";
+    options.TenantId += ValidScopeButNotTenantId;
+    AzureCliCredential azCliCred(options);
+
+    TokenRequestContext trc;
+    trc.Scopes.push_back(std::string("https://storage.azure.com/.default"));
+    EXPECT_THROW(static_cast<void>(azCliCred.GetToken(trc, {})), AuthenticationException);
+
+    try
+    {
+      auto const token = azCliCred.GetToken(trc, {});
+    }
+    catch (AuthenticationException const& e)
+    {
+      EXPECT_TRUE(std::string(e.what()).find("Unsafe") != std::string::npos) << e.what();
+    }
+  }
+
+  {
+    AzureCliCredentialOptions options;
+    options.CliProcessTimeout = std::chrono::hours(24);
+    AzureCliCredential azCliCred(options);
+
+    TokenRequestContext trc;
+    trc.Scopes.push_back(
+        std::string("https://storage.azure.com/.default") + ValidScopeButNotTenantId);
+
+    try
+    {
+      auto const token = azCliCred.GetToken(trc, {});
+    }
+    catch (AuthenticationException const& e)
+    {
+      EXPECT_TRUE(std::string(e.what()).find("Unsafe") == std::string::npos) << e.what();
```

Comment:
I'd at least find `"Unsafe command line input found in Scopes."`, if not the whole message including the credential prefix.

Reply:
The only reason I am doing so here, even partially, is to explicitly check that the exception is happening because of correct validation of unsupported characters and not because the artifact that we'd fail with AuthenticationException in CI for other reasons (even for valid tenant id/scopes). Otherwise, the test was passing even if we don't do any string validation.

My intention is not to verify exact string content of the message (which is an open topic to close on separately).

Ideally, I would prefer not to assert for any string literal.